### PR TITLE
Initial setup of review service

### PR DESCRIPTION
**Pull request by legendofa**
## DoD
- [x] Requirements of the issue are met
- [ ] Code has been reviewed
- [x] Code is documented
- [x] GraphQL related artefacts are documented

### CODE REVIEW

**Comment by legendofa** [comment]
The event endpoints will be converted in a separate PR. This PR can be tested without the gateway to check if the review endpoints match with what we are looking for.

Especially if the mutations behind the user resolver pose any problems.
Note: Some review endpoints are not yet resolved GraphQL entities for testing purposes.

File: base-dockerfile [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+RUN cargo chef cook --release --recipe-path recipe.json
+# Build application
+COPY . .
+
+RUN cargo build --release --bin misarch-review
+
+# We do not need the Rust toolchain to run the binary!
+FROM debian:bookworm-slim AS runtime
+
+RUN apt update && apt install -y protobuf-compiler && rm -rf /var/lib/apt/lists/*
```
Comment (nk-coding) [inline base-dockerfile]: just out of curiosity
with the dapr library gone
do we even need this anymore?

Comment (nk-coding) [reply base-dockerfile]: well I should have written this in the other PR...

File: src/mutation.rs [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+};
+
+/// Describes GraphQL review mutations.
+pub struct Mutation;
+
+#[Object]
+impl Mutation {
+    /// Entity resolver for user of specific id.
+    #[graphql(entity)]
+    async fn user_entity_resolver<'a>(
```
Comment (nk-coding) [inline src/mutation.rs]: having the entity resolver in the mutation thing looks a bit strange to me

Comment (nk-coding) [reply src/mutation.rs]: in particular as there is also a user_entity_resolver in query
do you need both?

**Review by nk-coding — CHANGES_REQUESTED** [review]
I don't know if you want to do it in this PR, but this service should also contribute an `averageReview` to Product
looking at it, we may want to rename it to `averageRating`

File: base-dockerfile (hunk 1 above)
Comment (legendofa) [reply base-dockerfile]: No, not necessary anymore, will remove this.

File: src/mutation.rs (hunk 2 above)
Comment (legendofa) [reply src/mutation.rs]: Yes, this is now unused code. Will fix

**Comment by legendofa** [comment]
> I don't know if you want to do it in this PR, but this service should also contribute an `averageReview` to Product looking at it, we may want to rename it to `averageRating`

Sounds like a good idea, i can do that in a separate PR.

**Comment by legendofa** [comment]
Well i pushed on the wrong branch, so this also include the GRPC -> HTTP conversion now. Sorry about that, i think this is easier than reverting a bunch of stuff.

**Review by nk-coding — APPROVED** [review]

**Review by delvh — APPROVED** [review]

**Pull request merged**